lib/redback.rb

```diff
@@ -31,6 +30,13 @@ def initialize(url, &each_site)
     spider
   end
 
+  def fix_url_formatting(url)
+    url = url.strip
+    url = "http://#{url}" unless url[/^http:\/\//] || url[/^https:\/\//]
+    url = "#{url}/" unless url.strip[-1] == "/"
```

Comment:
But the URL is already stripped (line 34), so it can't have leading or trailing whitespace at this point.